swaylock/main.c

```diff
@@ -156,7 +157,30 @@ static const struct wl_registry_listener registry_listener = {
 
 static struct swaylock_state state;
 
+bool create_file_lock() {
+	// Env var typically set by logind, e.g. "/run/user/<user-id>"
+	const char *dir = getenv("XDG_RUNTIME_DIR");
+	if (!dir) {
+		dir = "/tmp";
+	}
+	char path[256];
+	if (255 <= snprintf(path, 255,
+				"%s/swaylock.%i.pid", dir, getuid())) {
+		sway_abort("Lock path to long");
+	}
+	int pid_file = open(path, O_CREAT | O_RDWR, 0666);
+	int rc = flock(pid_file, LOCK_EX | LOCK_NB);
+	if(!rc) {
+		return true;
+	}
+	return false;
+}
+
 int main(int argc, char **argv) {
+	if (!create_file_lock()) {
```

Comment:
Bump: I still think this should be removed.

Reply:
If you remove this, the only difference will be that you will get
zwlr_input_inhibit_manager_v1@8: error 0: this compositor already has input inhibited

Since the scenario where two swaylock might be run at the same time, will be a quite common one with the swayidle trying to run swaylock in the background. I thought it would be better to handle this in a more controlled fashion. 
Can you please argue, the pros of having swaylock exit by
zwlr_input_inhibit_manager_v1@8: error 0: this compositor already has input inhibited

Reply:
Why does this happen? `swayidle` shouldn't spawn more than one `swaylock` right?